docs/troubleshooting.md

```diff
@@ -0,0 +1,73 @@
+# Troubleshooting
+
+If you have problems, try to rebuild when compiling with `-DCMAKE_BUILD_TYPE=Debug`.
+In particular also rebuild the index.
+The release build assumes machine written words- and docsfiles and omits sanity checks for the sake of speed.
+
+## Run End-to-End Tests
+
+QLever includes a simple mechanism for testing the entire system from
+from building an index to running queries in a single command.
+
+In fact this End-to-End Test is run on Travis CI for every commit and Pull
+Request. It is also useful for local development however since it allows you to
+quickly test if something is horribly broken.
+
+Just like QLever itself the End-to-End Tests can be executed from a previously
+build QLever container
+
+**Note**: If you encounter permission issues e.g. if your UID is not 1000 or you
+are using docker with user namespaces, add the flag `-u root` to your `docker
+run` command or do a `chmod -R o+rw e2e_data`
+
+    docker build -t qlever .
+    docker run -it --rm -v "$(pwd)/e2e_data:/app/e2e_data/" --name qlever-e2e --entrypoint e2e/e2e.sh qlever
+
+## Converting old Indices For Current QLever Versions
+
+We have recently updated the way the index meta data (offsets of relations
+within the permutations) is stored. Old index builds with 6 permutations will
+not work directly with the recent QLever version while 2 permutation indices
+will work but throw a warning at runtime.
+
+We have provided a converter which
+allows to only modify the meta data without having to rebuild the index. Just
+run `MetaDataConverterMain <index-prefix>` in the same way as the
+`IndexBuilderMain`.
+
+This will not automatically overwrite the old index but copy the permutations
+and create new files with the suffix `.converted` (e.g.
+`<index-prefix>.index.ops.converted` These suffixes have to be removed manually
+in order to use the converted index (rename to `<index-prefix>.index.ops` in our
+example).
+**Please consider creating backups of
+the "original" index files before overwriting them like this**.
+
+Please note that for 6 permutations the converter also builds new files
+`<index-prefix>.index.xxx.meta-mmap` where parts of the meta data of OPS and OSP
+permutations will be stored.
+
+
+## High RAM Usage During Runtime
+
+QLever uses an on-disk index and is usually able to operate with pretty low RAM
+usage. For example it can handle the full Wikidata KB + Wikipedia which is about
+1.5 TB of index with less than 46 GB of RAM
+
+However, there are data layouts that currently lead to an excessive
+amount of memory being used.
```

Comment:
Without more context this sentence is confusing. Which data layouts are the problem? Is there something a user should avoid doing?
If this references the list below, you should use a colon instead of a full stop at the end of the sentence.